commandlineparser/include/Parser.h

```diff
@@ -62,6 +64,48 @@ namespace cmdParser
 		//storing the tokenized data.		
 		std::vector<std::string> tokenized_data;
 	};
+
+	template <typename T>
+	void cmdParser::Parser::AddOptions(std::string short_command, std::string long_command, std::string short_description, std::string long_description, T def_value)
+	{
+		static_assert(!std::is_convertible<T, const char *>::value, "const char * data type for string literal is not supported in this version !");
+		
+		std::shared_ptr<cmdParser::Options> obj = std::make_shared <cmdParser::Options_with_value<T>>(short_command, long_command, short_description, long_description, def_value);
+
+		bool is_short_empty = false;
+
+		if (!(obj->get_option_short_command().empty()))
+		{
+			command_list.insert(KeyValue(obj->get_option_short_command(), obj));
+		}
+		else
+		{
+			is_short_empty = true;
+		}
+		if (!(obj->get_option_long_command().empty()))
+		{
+			command_list.insert(KeyValue(obj->get_option_long_command(), obj));
+		}
+		else if (is_short_empty)
+		{
+			throw std::exception("Enter a valid short or long command or both.");
+		}
+	}
+
+	template<typename U>
+	std::vector <U> cmdParser::Parser::getValueAsString(const std::string & input) const
+	{
+		auto ptr = dynamic_cast<Options_with_value<U>*>(command_list.at(input).get());
+
+		if (ptr != nullptr)
+		{
+			return ptr->get_value();
+		}
+		else
+		{
+			throw std::exception(std::string(std::string("Type mismatch for input argument ") + input).c_str());
```

Comment:
Add a descriptive error message  saying what T is passed and if possible also tell  for current option what is the  T required

Reply:
Will do in next feature branch.  (#25 User should be able to get value as defined data types in C++ enhancement )